src/app/components/client/UpsellDialog.tsx

```diff
@@ -76,6 +76,51 @@ function UpsellDialogContent({
   const [selectedTab, setSelectedTab] = useState<Key>(defaultSelectedKey);
   const recordTelemetry = useTelemetry();
 
+  // format subscription urls
```

Comment:
Am I correct in thinking that this logic is essentially duplicated in a couple of places? If so, might it make sense to extract it into a separate function in `src/app/functions/universal/`? Might also make it easier to understand what is actually being done (and why).

Reply:
Done